cosima_cookbook/querying.py

```diff
@@ -1,13 +1,73 @@
 import logging
 import os.path
+import pandas as pd
+from sqlalchemy import func, distinct
 
 import xarray as xr
 
 from . import database
 
+from .database import NCExperiment, NCFile, CFVariable, NCVar
+
 class VariableNotFoundError(Exception):
     pass
 
+def get_experiments(session):
+    """
+    Returns a dataframe all experiments and number of netCDF4 files
+    """
+
+    q = session.query(NCExperiment.experiment,
+              func.count(NCFile.experiment_id).label('ncfiles'))\
+            .group_by(NCFile.experiment_id).join(NCExperiment)
+
+    return pd.DataFrame(q)
+
+def get_ncfiles(session, experiment):
+    """
+    Returns a dataframe of all netcdf files for a given experiment
+    """
+
+    q = session.query(NCFile.ncfile, NCFile.index_time).join(NCExperiment)\
+        .filter(NCExperiment.experiment==experiment)
+
+    return pd.DataFrame(q)
+
+def get_variables(session, experiment, frequency=None):
+    """
+    Returns DataFrame of variables for a given experiment
+    """
+
+    q = session.query(CFVariable.name, NCFile.frequency,
+                  NCFile.ncfile,
```

Comment:
Additionally, I think it's clearer to indent these function arguments another level, so you can distinguish them from the query chain?

Reply:
ok